WordPress/src/test/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModelTest.kt

```diff
@@ -66,20 +74,134 @@ class ReaderInterestsViewModelTest {
             initViewModel()
 
             // Then
-            assertThat(viewModel.uiState.value).isNull()
+            assertThat(viewModel.uiState.value).isEqualTo(
+                UiState(
+                    listOf(),
+                    ReaderTagList(),
+                    DoneButtonHiddenUiState
+                )
+            )
+        }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `done button hidden on start switches to disabled state when tags received from repo`() =
+        coroutineScope.runBlockingTest {
+            // Given
+            val mockInterests = getMockInterests()
+            whenever(readerTagRepository.getInterests()).thenReturn(mockInterests)
+
+            // Pause dispatcher so we can verify done button initial state
+            coroutineScope.pauseDispatcher()
+
+            // Trigger data load
+            initViewModel()
+
+            assertThat(requireNotNull(viewModel.uiState.value).doneButtonUiState)
+                .isInstanceOf(DoneButtonHiddenUiState::class.java)
+
+            // Resume pending coroutines execution
+            coroutineScope.resumeDispatcher()
+
+            assertThat(requireNotNull(viewModel.uiState.value).doneButtonUiState)
+                .isInstanceOf(DoneButtonDisabledUiState::class.java)
+        }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `done button enabled if at least one interest selected`() =
+        coroutineScope.runBlockingTest {
+            // Given
+            val mockInterests = getMockInterests()
+            whenever(readerTagRepository.getInterests()).thenReturn(mockInterests)
+
+            // When
+            initViewModel()
+            viewModel.onInterestAtIndexToggled(index = 0, isChecked = true)
+
+            // Then
+            assertThat(requireNotNull(viewModel.uiState.value).interestsUiState.filter { it.isChecked }.size)
+                .isGreaterThanOrEqualTo(1)
+            assertThat(requireNotNull(viewModel.uiState.value).doneButtonUiState)
+                .isInstanceOf(DoneButtonEnabledUiState::class.java)
+            assertThat(requireNotNull(viewModel.uiState.value).doneButtonUiState.titleRes)
+                .isEqualTo(R.string.reader_btn_done)
+            assertThat(requireNotNull(viewModel.uiState.value).doneButtonUiState.enabled)
+                .isEqualTo(true)
+        }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `done button disabled if the only selected interest is deselected`() =
+        coroutineScope.runBlockingTest {
+            // Given
+            val mockInterests = getMockInterests()
+            whenever(readerTagRepository.getInterests()).thenReturn(mockInterests)
+
+            // When
+            initViewModel()
+            // Select an interest
+            viewModel.onInterestAtIndexToggled(index = 0, isChecked = true)
+            // Then
+            assertThat(requireNotNull(viewModel.uiState.value).interestsUiState.filter { it.isChecked }.size)
```

Comment:
I feel like these two assertions don't belong to this test. These assertions are already being tested in the test above -> each the test should ideally have a single purpose/focus/reasonToFail. Wdyt?

Reply:
Refactored here: a456645